### PR TITLE
python_examples: Reformatted, self-checking, executable

### PR DESCRIPTION
**Pull request by pylbert**
* Moved body of each python example to main.  This allows for basic
      load module testing for CI
    * General cleanup of python modules (crlf/tabs/prints/etc)
    * Chmod'ed to 755 to allow running examples without specifying the
      python interpreter
    * Added ctest for loading python2/3 modules
    * Added jniclasscode pragma for java swig interface files.
    * Updated check_examplenames.py module to check all languages vs. a
      cxx example name
    * Added tests for checking python module and test loading
    * Added 'make test' to travis-ci run (run ctests)
    * Print a more meaningful message when not building cxx docs into
      python modules
    * Updated check_clean.py to only check java wrapper files.
    * ENABLED ctests for UPM

Signed-off-by: Noel Eck <noel.eck@intel.com>

### CODE REVIEW

File: examples/python/using_carrays.py [hunk 1]
```diff
@@ -24,8 +24,11 @@
```
Comment (jontrulson) [inline examples/python/using_carrays.py]: Hmm, is it even worth keeping this example?

File: src/lp8860/javaupm_lp8860.i [hunk 2]
```diff
@@ -6,3 +6,14 @@
 %}
 
 %include "lp8860.hpp"
+
```
Comment (jontrulson) [inline src/lp8860/javaupm_lp8860.i]: Ideally these fixes would be a separate PR, but no biggie.

File: tests/check_examplenames.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+# Example name map
+sampleMappingFile = '../doxy/samples.mapping.txt'
+
+# Example directories
+sampledir = {}
+sampledir['cxx'] = '../examples/c++/'
+sampledir['java'] = '../examples/java/'
+sampledir['js'] = '../examples/javascript/'
+sampledir['py'] = '../examples/python/'
+
```
Comment (jontrulson) [inline tests/check_examplenames.py]: Should this handle C examples as well?

File: tests/check_pythonload.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+             # Requires mraa python module
+             'curieimu',
+             # Requires modbus
+             'h803x', 'hwxpxx', 't3311',
+             # Requires bacnet
+             'e50hx', 'tb7300', 't8100',
+             # Requires PIL
+             'make_oled_pic'
+             ]
+
```
Comment (jontrulson) [inline tests/check_pythonload.py]: Would it be better to have these in a datafile rather than hardcoded into a script?

File: doxy/samples.mapping.txt [hunk 5]
```diff
@@ -1,73 +1,72 @@
...
-stepmotor.cxx	StepMotorSample.java
-tm1637.cxx	TM1637Sample.java
-tp401.cxx	TP401Sample.java
-tsl2561.cxx	TSL2561Sample.java
-ttp223.cxx	TTP223Sample.java
-ublox6.cxx	Ublox6Sample.java
-uln200xa.cxx	ULN200XASample.java
-wt5001.cxx	WT5001Sample.java
-yg1006.cxx	YG1006Sample.java
+# Mapping examples across provided languages
```
Comment (jontrulson) [inline doxy/samples.mapping.txt]: Interesting, I was not aware of this file.  No doubt it is incomplete.  What about C examples?  How are cases handled where a driver has more than one example (urm37 vs. urm37-uart, for example).

File: examples/python/aeotecdsb09104.py [hunk 6]
```diff
@@ -24,65 +24,50 @@
...
-
-        print "Energy, Channel 3:",
-        print sensor.getEnergyC3(),
-        print "kWh"
-
-        print "Battery Level:",
-        print sensor.getBatteryLevel(),
-        print "%"
-        print
+        print "Battery Level: %0.03f\n" % sensor.getBatteryLevel()
```
Comment (pylbert) [inline examples/python/aeotecdsb09104.py]: Fixed...

File: examples/python/using_carrays.py (hunk 1 above)
Comment (pylbert) [inline examples/python/using_carrays.py]: Not sure what this was ever used for.  If it's just there to show how to use a swig carray type then I agree.

File: tests/check_pythonload.py (hunk 4 above)
Comment (pylbert) [inline tests/check_pythonload.py]: I would like to keep from adding additional files for testing.  One thought is to install the tests and examples according to the cmake options and run from the install directory.  In this way, if tpv libraries were not available during the cmake generation phase, then the installed examples would reflect this (and there would be no need for a blacklist in this case).

File: examples/python/using_carrays.py (hunk 1 above)
Comment (Propanu) [inline examples/python/using_carrays.py]: OK to remove, we have plenty of examples for this within the drivers that actually use it.

Comment (pylbert) [inline examples/python/using_carrays.py]: Removed...
